app/components/ui/Navigation/DropdownSubmenu.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import navStyles from "./DesktopNavigation.module.scss";
+import categoryStyles from "./CategoryDropdown.module.scss";
+
+// This component is the dropdown submenu used by both DesktopMenuItems and CategoryDropdown, created in service of DRY code
+
+const DropdownSubmenu = ({
+  title,
+  links,
+  activeSubMenu,
+  handleMenuToggle,
+  uniqueKey,
+  menuRef,
+  variant = "navigation",
+}: {
+  title: string;
+  links: { id: number | string; url: string; text: string }[];
+  activeSubMenu: string | null;
+  handleMenuToggle: (key: string | null) => void;
+  uniqueKey: string;
+  menuRef: React.RefObject<HTMLDivElement>;
+  variant?: "navigation" | "category";
+}) => {
```

Comment:
I feel like you're actually really close to an interface that is entirely decoupled from a variant type and would be generic for any reusable dropdown case if you can pass in a `styles` parameter:
```
styles: {
    [key: string]: string;
};
```